persist/sqlite/consensus_refactored_test.go

```diff
@@ -909,3 +1011,304 @@ func TestMissingBlock(t *testing.T) {
+			SiacoinOutput: genesisOutput,
+		},
+	})
+	checkSiacoinOutputs(addr2)
+	checkSiacoinOutputs(addr3)
+}
+
+func TestUnspentSiafundOutputs(t *testing.T) {
+	pk1 := types.GeneratePrivateKey()
+	uc1 := types.StandardUnlockConditions(pk1.PublicKey())
+	addr1 := uc1.UnlockHash()
+
+	pk2 := types.GeneratePrivateKey()
+	uc2 := types.StandardUnlockConditions(pk2.PublicKey())
+	addr2 := uc2.UnlockHash()
+
+	pk3 := types.GeneratePrivateKey()
+	uc3 := types.StandardUnlockConditions(pk3.PublicKey())
+	addr3 := uc3.UnlockHash()
+
+	n := newTestChain(t, false, func(network *consensus.Network, genesisBlock types.Block) {
+		genesisBlock.Transactions[0].SiafundOutputs[0].Address = addr1
+	})
+	genesisOutput := n.genesis().Transactions[0].SiafundOutputs[0]
+	genesisOutputID := n.genesis().Transactions[0].SiafundOutputID(0)
+
+	checkSiafundOutputs := func(addr types.Address, expected ...explorer.SiafundOutput) {
+		t.Helper()
+
+		sfos, err := n.db.UnspentSiafundOutputs(addr, 0, math.MaxInt64)
+		if err != nil {
+			t.Fatal(err)
+		}
+		testutil.Equal(t, "len(sfos)", len(expected), len(sfos))
+
+		for i := range sfos {
+			testutil.Equal(t, "SpentIndex", expected[i].SpentIndex, sfos[i].SpentIndex)
+			testutil.Equal(t, "SiafundOutput", expected[i].SiafundOutput, sfos[i].SiafundOutput)
+		}
+	}
+
+	// only addr1 should have SF from genesis block
+	checkSiafundOutputs(addr1, explorer.SiafundOutput{
+		SiafundElement: types.SiafundElement{
+			ID:            genesisOutputID,
+			SiafundOutput: genesisOutput,
+		},
+	})
+	checkSiafundOutputs(addr2)
+	checkSiafundOutputs(addr3)
+
+	txn1 := types.Transaction{
+		SiafundInputs: []types.SiafundInput{{
+			ParentID:         genesisOutputID,
+			UnlockConditions: uc1,
+		}},
+		SiafundOutputs: []types.SiafundOutput{{
+			Address: addr2,
+			Value:   genesisOutput.Value,
+		}},
+	}
+	testutil.SignTransaction(n.tipState(), pk1, &txn1)
+
+	txn2 := types.Transaction{
+		SiafundInputs: []types.SiafundInput{{
+			ParentID:         txn1.SiafundOutputID(0),
+			UnlockConditions: uc2,
+		}},
+		SiafundOutputs: []types.SiafundOutput{{
+			Address: addr3,
+			Value:   genesisOutput.Value,
+		}},
+	}
+	testutil.SignTransaction(n.tipState(), pk2, &txn2)
+
+	// net effect of txn1 and txn2 is to send addr1 output to addr3
+	n.mineTransactions(t, txn1, txn2)
+
+	// addr3 should have all the value now
+	checkSiafundOutputs(addr1)
+	checkSiafundOutputs(addr2)
+	checkSiafundOutputs(addr3, explorer.SiafundOutput{
+		SiafundElement: types.SiafundElement{
+			ID:            txn2.SiafundOutputID(0),
+			SiafundOutput: txn2.SiafundOutputs[0],
+		},
+	})
+
+	n.revertBlock(t)
+
+	// after revert, addr1 should have the output again and the others should
+	// have nothing
+	checkSiafundOutputs(addr1, explorer.SiafundOutput{
+		SiafundElement: types.SiafundElement{
+			ID:            genesisOutputID,
+			SiafundOutput: genesisOutput,
+		},
+	})
+	checkSiafundOutputs(addr2)
+	checkSiafundOutputs(addr3)
+}
+
+func TestBlock(t *testing.T) {
```

Comment:
What's the intention of this test? There's no solid assertions here. We would probably have to assert counts at the very least because we're just ranging over the blocks and seeing if they are present in the database, but those transactions don't really matter.

Reply:
To check that nonce/timestamp/miner payouts are set.  I simplified it a lot in recent commit should be better now.

Reply:
Perhaps we should pass an expected amount of blocks? We could do `checkBlocks(1) - mine - checkBlocks(2) - revert - checkBlocks(1)` perhaps? Fine to resolve if that doesn't make sense, my issue with the test is that `checkBlocks` just assumes `n.blocks` is being updated. Obviously that is the case but this test would still pass if we wouldn't append or remove from `n.blocks`